### DISCUSSION TITLE
Any way to make changing to stacking similar to changing to vertical/horizontal layout

### DISCUSSION THREAD

**Discussion by jam1015** [i3/i3#5588] [Q&A]
for background here are my relevant keybindings:

```
# start a terminal
bindsym $mod+Return exec kitty

# split in horizontal orientation
bindsym $mod+s split v

# split in vertical orientation
bindsym $mod+v split h

# enter fullscreen mode for the focused container
bindsym $mod+f fullscreen toggle

# change container layout (stacked, tabbed, toggle split)
bindsym $mod+w layout stacking
bindsym $mod+t layout tabbed
bindsym $mod+e layout toggle split

# focus the parent container
bindsym $mod+u focus parent

# focus the child container
bindsym $mod+i focus child

```

The inconsistent behavior:
Open a new workspace.

input `$mod+return` twice. I now have two terminals on either side of the screen.

In one case I run `$mod+s` and the only visual change is that the bottom of the selected window is highlighted. I run `$mod+return` again and that selected window is split with one new terminal physically underneath the selected terminal.

In the other case, I enter `$mod+t` and both original terminals are in a tab layout. To make the tab layout apply to a single of the original two terminals terminal I have to input `$mod+s` or `$mod+v` to make it a container, and then input `$mod+t` to get the original expected behavior of having the tabbed layout just apply to that terminal.  Then further presses of `$mod+return` open new tabs just in that container.  

The behavior I expect is to not have to press `$mod+t` or `$mod+v` beforehand, `$mod+t` should just apply to the selected terminal. Under the expected-behavior to make both of the original terminals tabs  I would have to input `$mod+u` (focus parent) to select both initial terminals, and then `$mod+t` .

Does anyone know how to make the behavior consistent?

**Comment by okraits** [comment]
Two annotations:

1. You can chain commands, so you can have a keybinding `bindsym $mod+t split v; layout tabbed` the desired behaviour with one key combination.
2. There's also `layout toggle` to cycle through a set of layouts: https://i3wm.org/docs/userguide.html#manipulating_layout
